gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/AsStringGlobalStep.java

```diff
@@ -32,15 +32,15 @@
  * @author David Bechberger (http://bechberger.com)
  * @author Yang Xia (http://github.com/xiazcy)
  */
-public final class AsStringStep<S> extends ScalarMapStep<S, String> {
+public final class AsStringGlobalStep<S, E> extends ScalarMapStep<S, E> {
```

Comment:
Why did you decide to change this type parameter from <S, String> to <S, E>?

Reply:
`public default GraphTraversal<S, E> asString(final Scope scope)`
This will complain otherwise, cause the local and global scopes return different thing (string vs list of string).